### PR TITLE
Move wrapPlayer out of useMemo in PlayerManager

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
None

**Description**
The `wrapPlayer` call runs in a memo which executes during the _render_ phase in react. However, `wrapPlayer` may run logic which invokes a state update. React does not like state updates of a component to occur when another component is rendering. This moves the construction of the wrapped player to a callback that builds the topicAliasPlayer and the wrapped player and then sets the state.

This avoids the error for updating a component while rendering.

### CODE REVIEW

**Review by jtbandes — COMMENTED** [review]
> wrapPlayer may run logic which invokes a state update

Why is that?

**Comment by defunctzombie** [comment]
> Why is that?

See AppContextProvider line 110.

File: packages/studio-base/src/components/PlayerManager.tsx [hunk 1]
```diff
@@ -53,17 +53,28 @@ export default function PlayerManager(props: PropsWithChildren<PlayerManagerProp
   const analytics = useAnalytics();
   const metricsCollector = useMemo(() => new AnalyticsMetricsCollector(analytics), [analytics]);
 
-  const [basePlayer, setBasePlayer] = useState<Player | undefined>();
+  const [playerState, setPlayerState] = useState<
```
Comment (jtbandes) [inline packages/studio-base/src/components/PlayerManager.tsx:56]: Would recommend renaming this since "PlayerState" is already a core concept in the app.

**Review by jtbandes — APPROVED** [review]
Approach seems fine. Another thing that I wondered about is whether we could move the side effects out of `wrapPlayer` and into `useEffect`s? Maybe this is more straightforward though.

**Pull request merged**